src/fhirtypes/ElementDefinition.ts

```diff
@@ -1889,6 +1892,77 @@ export class ElementDefinition {
     return [];
   }
 
+  /**
+   * Returns an array of an ElementDefinition's unfolded children.
+   * @param {ElementDefinition} targetElement - The ElementDefinition being unfolded
+   * @returns {ElementDefinition[]} An array of the targetElement's children, with the IDs altered and
+   * the original property re-captured.
+   */
+
+  private cloneChildren(targetElement: ElementDefinition): ElementDefinition[] {
+    return targetElement?.children().map(e => {
+      const eClone = e.clone();
+      eClone.id = eClone.id.replace(targetElement.id, this.id);
+      eClone.structDef = this.structDef;
+      eClone.captureOriginal();
+      return eClone;
+    });
+  }
+
+  /**
+   * Checks a StructureDefinition's differential to determine if the profile element extension has been used on
+   * an element.
+   * @param {Object} profileJson - The json representation of this ElementDefinition's structDef
+   * @returns {boolean} True if the profile element extension is found on this elements profile property, false
+   * if the extension is not found
+   */
+  private hasProfileElementExtension(profileJson: any): boolean {
+    const elementName = this.getContentReferenceId();
+
+    const elementType = profileJson.differential.element.find(
+      (element: any) => element.id === elementName
+    )?.type?.[0];
+
+    // If the type property is not present in the differential, the profile-element extension is not present
+    if (!elementType) {
+      return false;
+    }
+
+    let profileCanonical, extensionUrl, targetElement: string;
+
+    // If the profile and _profile arrays are not present, the profile-element extension is not present
+    if (elementType.profile && elementType._profile) {
+      const profileIndex = elementType._profile.findIndex(
+        (profile: any) =>
+          profile &&
+          profile.extension.some(
+            (extension: any) =>
+              extension.hasOwnProperty('url') && extension.hasOwnProperty('valueString')
+          )
+      );
```

Comment:
I think this will need to be adjusted, because if there's more than one extension with a string value on the element, the one that gets found might not be the profile element extension. I got the test case in `StructureDefinitionExporter.test.ts` on line 248 by having these rules on the profile:
```ts
    const otherExtensionRule = new CaretValueRule('item');
    otherExtensionRule.caretPath = 'type.profile.extension[0].url';
    otherExtensionRule.value = 'http://hl7.org/fhir/us/minimal/StructureDefinition/FooExtension';

    const otherExtensionValue = new CaretValueRule('item');
    otherExtensionValue.caretPath = 'type.profile.extension[0].valueString';
    otherExtensionValue.value = 'Foo';

    const extensionRule = new CaretValueRule('item');
    extensionRule.caretPath = 'type.profile.extension[1].url';
    extensionRule.value =
      'http://hl7.org/fhir/StructureDefinition/elementdefinition-profile-element';
    const targetElementRule = new CaretValueRule('item');
    targetElementRule.caretPath = 'type.profile.extension[1].valueString';
    targetElementRule.value = 'Questionnaire.item';
```

The function finds the first extension, which isn't the profile element extension, so the function returns false.

Reply:
This is totally right, thank you for finding this! In commit `96fa4a6` I've added in a third condition to the search on line 1940 that ensures that the extension's `url` property is the profile-element extension's canonical. That should address this case, and I've also added in a unit test that tests this scenario.